gateway-js/src/schema-helper/__tests__/addExtensions.test.ts

```diff
@@ -3,9 +3,31 @@ import { addExtensions } from '../addExtensions';
 const { version } = require('../../../package.json');
 
 describe('addExtensions', () => {
+
   it('adds gateway extensions to a schema', async () => {
     const schema = buildSchema('type Query { hello: String }');
     expect(schema.extensions).toEqual({});
     expect(addExtensions(schema).extensions).toEqual({ apollo: { gateway: { version: version } } });
   });
+
+  it('does not delete existing extensions', async () => {
```

Comment:
This test passes whether we're running with or without your fix. Did you want another test where no gateway was present?

Reply:
@clenfest I was trying to reproduce a failing test, but I can't set an invalid type to the `schema.extensions.apollo` without causing a TS compile error in our tests.

I am not sure why our TS compile step is not catching the current error in `main` because it is clearly an error if the type is `unknown`